docs/DeveloperGuide.md

```diff
@@ -1,38 +1,252 @@
+<!-- @@author yijiano -->
+
+PILL uses the following tools for development:
+1. [JUnit 5](https://junit.org/junit5/) - Used for testing.
+2. [Gradle](https://gradle.org/) - Used for build automation.
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The project is designed using the Model-View-Controller (MVC) architecture, with the following components:
+
+1. **Model**: Contains the core logic of the application, including the Item, ItemMap, and Storage classes.
+2. **View**: The user interface, which is implemented as a command-line interface (CLI).
+3. **Controller**: The command classes, which interpret user input and call the appropriate methods from the Model and
+   Storage classes.
+4. **Storage**: Handles the reading and writing of data to a CSV file.
+5. **DateTime**: A utility class to handle date and time operations.
+6. **Exceptions and Logging**: Custom exceptions and a logging utility class to manage application-wide logging.
+7. **PillLogger**: A centralized utility class to handle logging across the entire application.
+8. **PillException**: A custom exception class to handle exceptions specific to the PILL application.
+9. **Item**: A class to represent an item in the inventory.
+10. **ItemMap**: A class to store items in a map-like structure.
+11. **Command**: A class to represent a command that can be executed by the application.
+12. **Parser**: A class to parse user input and return the corresponding command.
+13. **Pill**: The main class that initializes the application and starts the CLI.
+
+### High-Level Overview
+
+The high-level overview of the project structure is as follows:
+
+<img src = "diagrams/High-Level-Overview.png" alt="High Level Overview of PILL"/>
+
+
+<!-- @@author cxc0418 -->
+
+### UI and I/O
+
+The program uses a command-line interface (CLI) for interaction with the user.
+It receives input as text commands, processes these commands, and provides feedback through the console.
+
+The UI consists of the following components:
+
+1. **Parser**: The Parser class is responsible for interpreting user input and returning the corresponding command.
+2. **Command**: The Command class represents a command that can be executed by the application.
+3. **Pill**: The main class that initializes the application and starts the CLI.
+
+The UI components work together to provide a seamless user experience, allowing users to interact with the application.
+
+### Commands
+
+Each user action (e.g. adding, deleting, or editing an item) is mapped to a specific command class.
+These classes handle the logic for interpreting the input and calling the appropriate methods from the ItemMap and
+Storage classes.
+
+Example:
+
+```
+AddItemCommand command = new AddItemCommand(itemName, quantity, expiryDate);
```

Comment:
I think this can be put in the implementation section, following the example they give us.